### PR TITLE
Add deploy logic to actions for gh-pages and custom domain

### PR DESCRIPTION
**Pull request by wagnst**
The logic how `gh-pages` is deployed is changed so that folder `/deploy` is deployed via actions step `peaceiris/actions-gh-pages@v3`. Also a `CNAME` is added.

### CODE REVIEW

File: deploy/index.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<html>
+<head>
+  <script>
+      var rawCsvUrl = 'https://raw.githubusercontent.com/lhsystems/technology-radar/master/Lufthansa%20Systems%20Tech%20Radar.csv';
+      window.location.href = 'https://radar.thoughtworks.com/?sheetId=' + encodeURIComponent(rawCsvUrl)
+  </script>
+</head>
+<body>
+</body>
+</html>
```
Comment (danielmohacsi) [inline deploy/index.html:11]: Missing newline here.

File: README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ![CSV validation](https://github.com/lhsystems/technology-radar/workflows/CSV%20validation/badge.svg)
 
-This repository contains the CSV file of our [technology radar](https://lhsystems.github.io/technology-radar/).
+This repository contains the CSV file of our [technology radar](https://radar.cnt.lhsystems.com).
```
Comment (danielmohacsi) [inline README.md:5]: 🏆

**Review by danielmohacsi — CHANGES_REQUESTED** [review]
Requesting change for the missing newline.

I've got no experience with this action that you're using, but shouldn't our index.html be on the `gh-pages` branch?

**Comment by wagnst** [comment]
@danielmohacsi the action does the git commit to `gh-pages` branch automatically, I took the logic it over from `landscape` repo.

**Comment by danielmohacsi** [comment]
@wagnst Ah, nice, thanks for the explanation!

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+name: Deploy
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (danielmohacsi) [inline .github/workflows/main.yml:11]: Doesn't this cause every pull request to be deployed, even before merging?

Comment (wagnst) [reply .github/workflows/main.yml:11]: I had the same question, raised it to @mwilmes in chat. My assumption was we should remove L10+11, also in `landscape` repo.

Comment (wagnst) [reply .github/workflows/main.yml]: removed L10-11 after quick check with @mwilmes here as well.

File: deploy/index.html (hunk 1 above)
Comment (wagnst) [reply deploy/index.html:11]: can you maybe add a editorconfig file to do this? vscode seems to remove newline by default... :-(
https://editorconfig.org/

Comment (danielmohacsi) [reply deploy/index.html]: let's not mix this activity with your PR, but I'm open to that.

**Review by danielmohacsi — APPROVED** [review]

File: deploy/index.html (hunk 1 above)
Comment (wagnst) [reply deploy/index.html]: newline is added. issue for editorconfig created.

**Pull request merged**